--- ./src/test/java/test2/Terminal$StringValue.java	2012-09-18 09:10:43.000000000 +0100
+++ ../megaco-abnf/src/main/java/mofokom/megaco/resource/impl/codec/abnf/Terminal$StringValue.java	2012-09-18 09:17:54.000000000 +0100
@@ -3,14 +3,15 @@

 import java.util.ArrayList;
+import java.util.regex.Pattern;
 
 public class Terminal$StringValue extends Rule
 {
@@ -19,6 +20,7 @@
     super(spelling, rules);
   }
 
+  static Pattern name = Pattern.compile("[a-wA-Wy-zY-Z]+");
   public static Terminal$StringValue parse(
     ParserContext context, 
     String regex)
@@ -30,11 +32,17 @@
     Terminal$StringValue stringValue = null;
     try
     {
       String value = 
         context.text.substring(
           context.index, 
           context.index + regex.length());
-
+      //System.out.println("" + regex +  " "  + context.text.charAt(context.index + regex.length()) + " " + value);
+        if(name.matcher(regex).matches() && name.matcher(String.valueOf(context.text.charAt(context.index + regex.length()))).matches()) {
+            return null;
+        }
+      
       if ((parsed = value.equalsIgnoreCase(regex)))
       {
         context.index += regex.length();
